pkg/cache/cache_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build unit || !integration
+
+package cache
```

Comment:
nit: slight preference for tests live in their own package (`cach_test`) when access to the internals of the "thing" being tested are not required.